javalib/src/main/scala/java/util/ArrayList.scala

```diff
@@ -0,0 +1,84 @@
+package java.util
+
+import scala.scalajs._
+
+class ArrayList[E] private (
+    inner: js.Array[Box[E]]
+  ) extends AbstractList[E]
+    with RandomAccess
+    with Cloneable
+    with Serializable { self =>
+
+  def this() {
+    this(new js.Array[Box[E]])
+  }
+
+  def this(initialCapacity: Int) {
+    this(new js.Array[Box[E]])
+  }
+
+  def this(c: Collection[_ <: E]) {
+    this(new js.Array[Box[E]])
+    addAll(c)
+  }
+
+  override def clear(): Unit =
+    inner.clear()
+
+  override def add(e: E): Boolean = {
+    inner.push(Box(e))
+    true
+  }
+
+  override def add(index: Int, element: E): Unit = {
+    checkIndexOnBounds(index)
+    inner.splice(index, 0, Box(element))
+  }
+
+  def get(index: Int): E = {
+    checkIndexInBounds(index)
+    inner(index).inner
+  }
+
+  def size(): Int =
+    inner.length
+
+  override def set(index: Int, element: E): E = {
+    val e = get(index)
+    inner.update(index, Box(element))
+    e
+  }
+
+  override def remove(index: Int): E = {
+    val e = get(index)
+    inner.splice(index, 1)
+    e
+  }
+
+  override def subList(fromIndex: Int, toIndex: Int): List[E] =
+    new ArrayList(inner.jsSlice(fromIndex, toIndex))
+
+  override def clone(): AnyRef =
+    new ArrayList(inner.jsSlice(0, size))
+
+  @inline
+  def ensureCapacity(minCapacity: Int) {
```

Comment:
`@inline` is not necessary. Empty methods are always inlined.

Also, the `: Unit =` is missing.